### PR TITLE
Lmangall/web 164 decide about precedence of host check and method check

### PR DESCRIPTION
**Pull request by lmangall**
In this branch the code was adapted to take a URI like [www.example.com](http://www.example.com/) and behave as a “real” web server, There is a little if/else that allows us to test from both the browser and a tester (without localhost)  It’s also a cleanup PR where I got rid of some stuff

### CODE REVIEW

**Comment by 552020** [comment]
I merged main in this branch, and solved the conflicts and I pushed to your branch @lmangall! This can be merged I would say.

**Review by 552020 — APPROVED** [review]
LGTM

**Comment by 552020** [comment]
@dantol29 you can review and eventually approve!

**Comment by 552020** [comment]
New conflicts ...

**Comment by 552020** [comment]
This PR still waits for @dantol29 approval. And it has some conflicts with main: @lmangall.

**Comment by lmangall** [comment]
Ready to be merged @dantol29 @552020

**Review by 552020 — DISMISSED** [review]
LGTM!

File: tests/tests_1.cpp
Comment (dantol29) [inline tests/tests_1.cpp:1]: Why are they commented out?

Comment (552020) [reply tests/tests_1.cpp:1]: Probably cause he had to solve some issue and then forgot to comment them in again. tests_1.cpp is kind of a legacy one (we could also delete one). The most advance one is tests.cpp. Anyway the branch has conflicts with the new main, so we need to reapprove it again after @lmangall makes the necessary changed.

**Review by 552020 — APPROVED** [review]
LGTM!

**Review by dantol29 — APPROVED** [review]
NICE!

**Pull request merged**